zap/src/main/java/org/parosproxy/paros/extension/history/ExtensionHistory.java

```diff
@@ -463,26 +464,37 @@ public void run() {
         }
     }
 
-    private void searchHistory(HistoryFilter historyFilter) {
+    /**
+     * Returns a sorted list of History IDs for the visible history references.
+     *
+     * @return a list of History IDs for the visible history references.
+     * @since 2.13.0
+     */
+    public List<Integer> getHistoryIds() {
         Session session = getModel().getSession();
 
+        try {
+            List<Integer> list =
+                    getModel()
+                            .getDb()
+                            .getTableHistory()
+                            .getHistoryIdsOfHistType(
+                                    session.getSessionId(),
+                                    HistoryReference.TYPE_PROXIED,
+                                    HistoryReference.TYPE_ZAP_USER,
+                                    HistoryReference.TYPE_PROXY_CONNECT);
+            Collections.sort(list);
```

Comment:
How will pulling all of the relevant Ids scale is the session is very large?
Should we be sorting in the db query instead of in the code?
These comments may well have applied to the previous PR, sorry ;)

Reply:
We think this is okay. It's already done elsewhere in code and we do allow for the whole table to be sorted.